storage/backup/backup-cli/src/backup_types/state_snapshot/restore.rs

```diff
@@ -170,11 +169,12 @@ impl StateSnapshotRestoreController {
         let futs_iter = chunks.into_iter().enumerate().map(|(chunk_idx, chunk)| {
             let storage = storage.clone();
             async move {
-                let (blobs, proof) = try_join!(
-                    Self::read_state_value(&storage, chunk.blobs.clone()),
-                    storage.load_bcs_file(&chunk.proof)
-                )?;
-                Result::<_>::Ok((chunk_idx, chunk, blobs, proof))
+                tokio::spawn(async move {
+                    let blobs = Self::read_state_value(&storage, chunk.blobs.clone()).await?;
+                    let proof = storage.load_bcs_file(&chunk.proof).await?;
+                    Result::<_>::Ok((chunk_idx, chunk, blobs, proof))
+                })
+                .await?
```

Comment:
Why not just spawn instead of the outside `async move`?  but of course I could be missing something

Reply:
to `?` the join handle.. so it's `Result<_>` instead of `Result<Result<_>>`